Tools/cases_generator/tier2_abstract_generator.py

```diff
@@ -0,0 +1,235 @@
+"""Generate the cases for the tier 2 redundancy eliminator/abstract interpreter.
```

Comment:
Please add some notes about this module to `Tools/cases_generator/README.md`  :)